### PR TITLE
Inmarsat L band

### PR DESCRIPTION
**Pull request by rpatel3001**
first cut at L band stuff for early review

seems to work with a fresh DB but needs some code to migrate old versions

currently just assumes an acars like JSON format inside the "imsl" key, but I'll probably change it soon to accept something closer to the satdump raw output, and also add support to acars_router

### CODE REVIEW

**Comment by fredclausen** [comment]
I'll take a longer look at it once you mark it ready for review, but I love this! Thank you for taking the time on this.

Some notes for your consideration

* Since this is a bespoke format, I wouldn't work too hard at trying to match it to the existing ones. Obviously, internal to ACARS Hub it uses acarsdec as the standard, so you will have to translate it but you only really have to do that in one place. I would suggest using a top level key like HFDL and dumpvdl2 do, where we can immediately know the format based on the key itself. You mentioned it in your comment, so it sounds like you're on the same page.
* For router to validate, it should end up in the `acars_vdlm2_parser` library. I don't know how much you know/want to know about rust, but I'm happy to work on that once you have a format in place.
* Router will also have to be told how to accept L band data, and how to feed it. Really it'll be just adding some config options, but again, I'm happy to help sort that out.
* ACARS Hub is really, really bad at displaying extra metadata that isn't present in the acarsdec format. I made an unfortunate choice when ACARS Hub was a baby app and went with something that is rather inflexible. That's why, if you look at the HFDL code paths (and maybe I added it to the dumpvdl2 code paths too? I forget) you'll see CPDLC stuff (and some other HFDL random things) dumped out as raw json and saved in the `libacars` column of the database. It was the easiest way of doing something quick and dirty to at least display the data. I didn't want to expand the functionality because eventually someday ACARS Hub will be rewritten and a much more robust db structure will replace it.
* I'd have to look harder at your code, but migration of the database is possible. I try to avoid it, but it's easy enough. That said, I suspect the issue is with the RRD database, and I'd have to look but I know I did an RRD in place upgrade to support HFDL.

**Comment by rpatel3001** [comment]
Thanks, RRD update code added but not tested.

I was initially planning to have a top level key, but satdump actually outputs an identifier in the JSON so the format can be known with just a little extra work. Not wrapping in a new key would let people use baremetal satdump instead of forcing the container on them, because I wouldn't need to reformat things at all.

Here's an example of the output: https://gist.github.com/rpatel3001/f54c8c81724ad83085ab6ea5433c8d8f. It would be just as easy to test for msg["source"]["app"]["name"] = "SatDump" instead of checking for msg["imsl"]. Thoughts?

**Comment by rpatel3001** [comment]
ready to go i think, pending your approval of no top level key. ended up fixing a bunch of small things also. this PR includes #1133 btw, but that one is much smaller and could be merged before reviewing this big one.

**Comment by fredclausen** [comment]
> I was initially planning to have a top level key, but satdump actually outputs an identifier in the JSON so the format can be known with just a little extra work. Not wrapping in a new key would let people use baremetal satdump instead of forcing the container on them, because I wouldn't need to reformat things at all.
> 
> Here's an example of the output: https://gist.github.com/rpatel3001/f54c8c81724ad83085ab6ea5433c8d8f. It would be just as easy to test for `msg["source"]["app"]["name"] = "SatDump"` instead of checking for `msg["imsl"]`. Thoughts?

This is perfectly fine with me. I'll review in a bit.

File: README.md
Comment (fredclausen) [inline README.md:1]: Awesome job cleaning this up. I had no idea I was that behind on the docs....

File: rootfs/webapp/acars_formatter.py [hunk 1]
```diff
@@ -94,6 +100,9 @@ def format_imsl_message(unformatted_message):
         if "ges_id" in sigunit:
             imsl_message["fromaddr"] = sigunit["ges_id"]
 
+        if "msgno" in sigunit:
+            imsl_message["msgno"] = sigunit["ref_no"]
```
Comment (rpatel3001) [inline rootfs/webapp/acars_formatter.py:104]: Marking to fix later

**Pull request merged**